### PR TITLE
Shopper app/accounts page

### PR DESCRIPTION
**Pull request by weifengtan**
- Created a popover when hovering over the name to display list of stuff to press
- Created Account page 
- Created a Coming Soon page for things that are coming soon

Screenshot 2024-09-02 at 9 53 35 PM

Screenshot 2024-09-02 at 9 55 02 PM

Screenshot 2024-09-02 at 9 55 41 PM

Screenshot 2024-09-02 at 9 55 57 PM

### CODE REVIEW

File: shopper-app/src/views/ComingSoon.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import TopBar from "@/components/TopBar"
+
+
+export function ComingSoon() {
+  return(
+    <div style={{ display: 'flex', justifyContent: "center", flexDirection: 'column'}}> 
+      <TopBar></TopBar>
+      <div style={{display: 'flex', justifyContent: 'center', flexDirection: 'column', alignItems: 'center'}}>
+        <h1> COMING SOON </h1>
```
Comment (6YuQing6) [inline shopper-app/src/views/ComingSoon.tsx:9]: translation

File: shopper-app/src/components/AccountCardBox.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+export default function OutlinedCard({ title, list }) {
+  const pageContext = React.useContext(PageContext);
+  const router = useRouter();
+  const handleNavigation = (page) => {
+    switch (page) {
+      case 'account':
+        pageContext.setPage('account')
+        router.push('/')
+        break;
```
Comment (nochoy) [inline shopper-app/src/components/AccountCardBox.jsx:17]: account is no longer SPA

File: shopper-app/src/components/SignInButton.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        <CustomButton
+          label='Sign-In-Button'
+          onClick={handleSignIn}
+          sx={{
+            width: '200px',
+            height: '25px',
+            padding: '12px',
+          }}
+        >
+          Sign In
```
Comment (6YuQing6) [inline shopper-app/src/components/SignInButton.tsx:153]: translation

File: shopper-app/src/components/AccountCardBox.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import Typography from '@mui/material/Typography';
+import { PageContext } from '@/context/Page';
+import { useRouter } from 'next/router';
+
+export default function OutlinedCard({ title, list }) {
```
Comment (nochoy) [inline shopper-app/src/components/AccountCardBox.jsx:9]: name of file and default export should be the same name. Also, `src/components/AccountCards.jsx` also has  a default export with the same name. Someone who has not worked on this code should know what these components are based off of their name

File: shopper-app/src/components/SignOutButton.tsx [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+            sx={{
+              marginLeft: 2,
+              width: 35,
+              height: 35,
+              marginRight: 2,
+            }}
+          />
+          <Box> 
+            <Typography> {loginContext.userName} </Typography>
+            <Typography color="text.secondary"> Account holder </Typography>
```
Comment (6YuQing6) [inline shopper-app/src/components/SignOutButton.tsx:188]: translation

File: shopper-app/src/components/SignOutButton.tsx [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+      {makeList(leftSideListArrBottom)} 
+    </div>
+  )
+
+  const popoverContentBoxRight = (
+    <div>
+      <h1 style={{ fontSize: '17px', fontWeight: 'bold' }}>Your Account</h1>
+      {makeList(rightSideListArr)} 
+      <div className="link-container">
+        <a onClick={handleSignOut} className='link-no-underline'> Sign Out </a>
```
Comment (6YuQing6) [inline shopper-app/src/components/SignOutButton.tsx:135]: translate

File: shopper-app/src/components/SignOutButton.tsx [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+      <h1 style={{ fontSize: '17px', fontWeight: 'bold' }}> Your Lists </h1>
+      {makeList(leftSideListArrTop)} 
+      <Divider sx={{ mt: '10px', borderWidth: '1px', mb: '10px'}}></Divider>
+      {makeList(leftSideListArrBottom)} 
+    </div>
+  )
+
+  const popoverContentBoxRight = (
+    <div>
+      <h1 style={{ fontSize: '17px', fontWeight: 'bold' }}>Your Account</h1>
```
Comment (6YuQing6) [inline shopper-app/src/components/SignOutButton.tsx:132]: translate

File: shopper-app/src/components/TopBar.tsx [hunk 8]
```diff
@@ -310,77 +311,77 @@ export default function TopBar() {
...
+//				</Box>
+//				<Box sx={{ display: 'flex', height: '50%', width: 'auto', alignItems: 'flex-start', fontSize: '1em' }}>
+//					<Typography variant='body2' noWrap fontWeight='bold' lineHeight='1' letterSpacing='0.035em'>
+//						{t("topbar.Account")}
+//					</Typography>
+//					<ArrowDropDownIcon sx={{ height: '15px', width: '18px', color: '#a7acb2' }} />
+//				</Box>
+//			</Box>
+//		</CustomButton>
+//	)
```
Comment (6YuQing6) [inline shopper-app/src/components/TopBar.tsx:384]: remove comments

File: shopper-app/src/views/Account.tsx
Comment (6YuQing6) [inline shopper-app/src/views/Account.tsx:1]: add translation

File: shopper-app/src/components/AccountCards.jsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        {title}
+      </Typography>
+      <Typography variant="body2" color="text.secondary">
+        {description}
+      </Typography>
+    </Box>
+  );
+};
+
+export default function OutlinedCard({ image, alt, title, description, onClick }) {
```
Comment (6YuQing6) [inline shopper-app/src/components/AccountCards.jsx:35]: make mouse into pointer onclick

File: shopper-app/src/components/SignInButton.tsx [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+import { PageContext } from '@/context/Page';
+import { LoginContext } from '@/context/Login';
+import { on } from 'events';
+
+export default function SignInButton() {
+  const { t } = useTranslation('common');
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const pageContext = React.useContext(PageContext);
+  const loginContext = React.useContext(LoginContext);
+  //pageContext.setPage()
```
Comment (nochoy) [inline shopper-app/src/components/SignInButton.tsx:18]: remove

File: shopper-app/src/components/SignInButton.tsx [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleNavigation = (page: string) => {
+    switch (page) {
+      case 'account':
+        router.push('/account') 
+        break;
```
Comment (nochoy) [inline shopper-app/src/components/SignInButton.tsx:38]: account is no longer SPA

File: shopper-app/src/components/SignInButton.tsx [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+    { name: 'Content & Devices', onClick: 'comingSoon' },
+    { name: 'Subscribe & Save Items', onClick: 'comingSoon' },
+    { name: 'Memberships & Subscriptions', onClick: 'comingSoon' },
+    { name: 'Prime Membership', onClick: 'comingSoon' },
+    { name: 'Amazon Credit Cards', onClick: 'comingSoon' },
+    { name: 'Music Library', onClick: 'comingSoon' },
+    { name: 'Start a Selling Account', onClick: 'comingSoon' },
+    { name: 'Register for a free Business Account', onClick: 'comingSoon' },
+    { name: 'Customer Service', onClick: 'comingSoon' }
+  ];
```
Comment (nochoy) [inline shopper-app/src/components/SignInButton.tsx:83]: translations

File: shopper-app/src/components/TopBar.tsx [hunk 13]
```diff
@@ -310,77 +311,77 @@ export default function TopBar() {
...
-				<Box sx={{ display: 'flex', height: '50%', width: 'auto', alignItems: 'flex-start', fontSize: '1em' }}>
-					<Typography variant='body2' noWrap fontWeight='bold' lineHeight='1' letterSpacing='0.035em'>
-						{t("topbar.Account")}
-					</Typography>
-					<ArrowDropDownIcon sx={{ height: '15px', width: '18px', color: '#a7acb2' }} />
-				</Box>
-			</Box>
-		</CustomButton>
-	)
+	//const signInButton = (
```
Comment (6YuQing6) [inline shopper-app/src/components/TopBar.tsx:314]: remove comments

File: shopper-app/src/components/SignInButton.tsx [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+    {name: 'Your Lists', onClick: 'comingSoon'},
+    {name: 'Create a List', onClick: 'comingSoon'}, 
+    {name: 'Find a List or Registry', onClick: 'comingSoon'}
+  ]
+  const open = Boolean(anchorEl);
+  const id = open ? 'sign-in-popover' : undefined;
+
+  const popoverContentBoxLeft = (
+    <div>
+    <h1 style={{ fontSize: '17px', fontWeight: 'bold' }} > Your Lists </h1>
```
Comment (nochoy) [inline shopper-app/src/components/SignInButton.tsx:95]: translations

File: shopper-app/src/components/SignInButton.tsx [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+    <h1 style={{ fontSize: '17px', fontWeight: 'bold' }}> Your Account </h1>
+     {makeList(rightSideArray)} 
+    </div>
+  )
+
+  const popoverContentBox = (
+    <Box
+      sx={{
+        display: 'flex',
+        marginTop: 'px'
```
Comment (nochoy) [inline shopper-app/src/components/SignInButton.tsx:111]: change to 0

Comment (nochoy) [reply shopper-app/src/components/SignInButton.tsx:111]: or remove

**Review by 6YuQing6 — CHANGES_REQUESTED** [review]

File: shopper-app/src/components/SignOutButton.tsx [hunk 16]
```diff
@@ -0,0 +1,262 @@
...
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'sign-in-popover' : undefined;
+
+  const handleNavigation = (page: string) => {
+    switch (page) {
+      case 'account': 
+        router.push('/account') 
+        break;
```
Comment (nochoy) [inline shopper-app/src/components/SignOutButton.tsx:59]: account is no longer SPA

File: shopper-app/src/components/AccountCards.jsx (hunk 9 above)
Comment (weifengtan) [reply shopper-app/src/components/AccountCards.jsx:35]: no

File: shopper-app/src/components/SignOutButton.tsx [hunk 17]
```diff
@@ -0,0 +1,262 @@
...
+  const handleSignOut = () => {
+    loginContext.setUserName('');
+    loginContext.setAccessToken('');
+    loginContext.setId('');
+    loginContext.setRole('');
+  } 
+
+  const handleSignIn = () => {
+    router.push('/login')
+  }
```
Comment (nochoy) [inline shopper-app/src/components/SignOutButton.tsx:42]: not used?

File: shopper-app/src/components/TopBar.tsx (hunk 8 above)
Comment (nochoy) [inline shopper-app/src/components/TopBar.tsx:384]: remove

File: shopper-app/src/styles/links.css
Comment (nochoy) [inline shopper-app/src/styles/links.css:1]: remove this

File: shopper-app/src/pages/index.tsx [hunk 18]
```diff
@@ -25,6 +28,8 @@ export default function Index() {
       {pageContext.page === 'cart' && <Cart />}
       {pageContext.page === 'checkout' && <Checkout />}
       {pageContext.page === 'orderHistory' && <OrderHistory />}
+      {pageContext.page === 'account' && <Account/>}
```
Comment (nochoy) [inline shopper-app/src/pages/index.tsx:31]: account is no longer SPA

File: shopper-app/src/pages/_app.tsx [hunk 19]
```diff
@@ -1,5 +1,6 @@
 import "../styles/globals.css";
 import "../styles/fonts.css";
+import "../styles/links.css"
```
Comment (nochoy) [inline shopper-app/src/pages/_app.tsx:3]: remove

File: shopper-app/src/views/Account.tsx [hunk 20]
```diff
@@ -0,0 +1,326 @@
...
+import { Divider } from "@mui/material";
+import AccountCardBox from "@/components/AccountCardBox"
+
+export function Account() {
+  const pageContext = React.useContext(PageContext);
+  const router = useRouter();
+
+  const title = () => {
+    return (
+      <h1> Your Account </h1>
```
Comment (nochoy) [inline shopper-app/src/views/Account.tsx:16]: translation

File: shopper-app/src/views/Account.tsx [hunk 21]
```diff
@@ -0,0 +1,326 @@
...
+    )
+  }
+
+  const handleNavigation = (page: string) => {
+    switch (page) {
+      case 'account':
+        pageContext.setPage('account')
+        router.push('/')
+        break;
```
Comment (nochoy) [inline shopper-app/src/views/Account.tsx:25]: account is longer SPA

File: shopper-app/src/views/Account.tsx [hunk 22]
```diff
@@ -0,0 +1,326 @@
...
+        <Grid item xs={12} sm={6} md={4}>
+          <AccountCards
+            title="Your Messages"
+            image={"https://m.media-amazon.com/images/G/01/x-locale/cs/help/images/gateway/self-service/fshub/9_messages._CB654640573_.jpg"}
+            description="Manage your account security settings"
+            alt="Messages"
+            onClick={() => handleNavigation('comingSoon')}
+          />
+        </Grid>
+      </Grid>
```
Comment (nochoy) [inline shopper-app/src/views/Account.tsx:159]: 150+ lines of repeated code. Try putting this stuff into a list like your other code

File: shopper-app/src/views/Account.tsx [hunk 23]
```diff
@@ -0,0 +1,326 @@
...
+            list = {otherPrograms}
+          />
+        </Grid>
+        <Grid item xs={12} sm={4}>
+          <AccountCardBox
+            title="Manage your data"
+            list = {manageYourData}
+          />
+        </Grid>
+      </Grid>
```
Comment (nochoy) [inline shopper-app/src/views/Account.tsx:295]: repeated code

File: shopper-app/src/views/ComingSoon.tsx [hunk 24]
```diff
@@ -0,0 +1,14 @@
+import TopBar from "@/components/TopBar"
+
+
+export function ComingSoon() {
+  return(
+    <div style={{ display: 'flex', justifyContent: "center", flexDirection: 'column'}}> 
+      <TopBar></TopBar>
```
Comment (nochoy) [inline shopper-app/src/views/ComingSoon.tsx:7]: <TopBar/>

File: shopper-app/src/views/ComingSoon.tsx (hunk 1 above)
Comment (nochoy) [inline shopper-app/src/views/ComingSoon.tsx:9]: translation

File: shopper-app/src/views/ComingSoon.tsx [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+      <div style={{display: 'flex', justifyContent: 'center', flexDirection: 'column', alignItems: 'center'}}>
```
Comment (nochoy) [inline shopper-app/src/views/ComingSoon.tsx:8]: use MUI's box instead of div

**Review by nochoy — CHANGES_REQUESTED** [review]
Missing translations, using vanilla HTML elements + CSS instead of MUI, lacking mobile responsiveness

File: shopper-app/src/components/AccountCardBox.jsx [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+
+  const mappedOutLinks = (arr) => {
+    return (
+      <div>
+        {
+          arr.map((item, index) => (
+            <div key={index} className="link-container">
+              <a onClick={() => handleNavigation(item.onClick)} className='link-no-underline-blue'>
+                {item.name}
+              </a>
```
Comment (nochoy) [inline shopper-app/src/components/AccountCardBox.jsx:39]: use custom link component

File: shopper-app/src/views/ComingSoon.tsx (hunk 24 above)
Comment (weifengtan) [reply shopper-app/src/views/ComingSoon.tsx]: topbar tag

File: shopper-app/src/components/SignOutButton.tsx [hunk 27]
```diff
@@ -0,0 +1,261 @@
...
+        height: '60px',
+        border: 'none',
+        '&:hover': {
+          backgroundColor: '#131921',
+          outline: '1px solid white',
+          borderRadius: '2px',
+          border: 'none'
+        },
+      }}
+      onClick={handleSignOut}
```
Comment (nochoy) [inline shopper-app/src/components/SignOutButton.tsx:215]: onClick should go to account page

File: shopper-app/src/components/SignInButton.tsx [hunk 28]
```diff
@@ -0,0 +1,217 @@
...
+      {popoverContentBox}
+    </Box>
+  );
+
+  return (
+    <Box sx={{ height: '60px', display: 'flex', alignItems: 'center' }}>
+      <Button
+        aria-describedby={id}
+        onMouseEnter={handleOpen}
+        onClick={handleOpen}
```
Comment (nochoy) [inline shopper-app/src/components/SignInButton.tsx:166]: onClick should go to account page

**Review by nochoy — CHANGES_REQUESTED** [review]
- [ ] remove bold styling from "Your Account" and logo card titles 
- [ ] get correct gray text color from Amazon
- [ ] Prime logo cutoff
- [ ] Fix font size for popover
- [ ] Fix these borders
image

- [ ] 
- [ ] 
- [ ] mobile responsiveness

File: shopper-app/src/components/SignInButton.tsx [hunk 29]
```diff
@@ -0,0 +1,258 @@
...
+    { name: t('sign-in-button.left-side-array.your-lists'), onClick: 'comingSoon' },
+    { name: t('sign-in-button.left-side-array.create-a-list'), onClick: 'comingSoon' },
+    { name: t('sign-in-button.left-side-array.find-a-list-or-registry'), onClick: 'comingSoon' }
+  ];
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout;
+    if (isHovering) {
+      timer = setTimeout(() => {
+        console.log('hover')
```
Comment (6YuQing6) [inline shopper-app/src/components/SignInButton.tsx:112]: remove

File: shopper-app/src/components/SignInButton.tsx [hunk 30]
```diff
@@ -0,0 +1,258 @@
...
+      onMouseLeave={() => {
+        setIsHovering(false);
+        handleClose(); // Close popover when mouse leaves
+      }}   
+    >
+      <Button
+        id='sign-in-button'
+        aria-describedby={id}     
+        onClick={(e) => {
+          console.log('clicked')
```
Comment (6YuQing6) [inline shopper-app/src/components/SignInButton.tsx:205]: remove

File: shopper-app/src/components/SignOutButton.tsx [hunk 31]
```diff
@@ -0,0 +1,318 @@
...
+    { name: t('sign-out-button.left-side-arr-bottom.create-a-list'), onClick: 'comingSoon' },
+    { name: t('sign-out-button.left-side-arr-bottom.find-a-list-or-registry'), onClick: 'comingSoon' },
+    { name: t('sign-out-button.left-side-arr-bottom.your-saved-books'), onClick: 'comingSoon' }
+  ];
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout;
+    if (isHovering) {
+      timer = setTimeout(() => {
+        console.log('hover')
```
Comment (6YuQing6) [inline shopper-app/src/components/SignOutButton.tsx:144]: remove

File: shopper-app/src/components/SignOutButton.tsx [hunk 32]
```diff
@@ -0,0 +1,318 @@
...
+  );
+
+  const usernameButton  = (
+    <CustomButton
+      style={customButtonStyles}
+      label='sign-out-button'
+      id = 'sign-out-button'
+      variant='text'
+      onClick={(e) => {
+        console.log('clicked')
```
Comment (6YuQing6) [inline shopper-app/src/components/SignOutButton.tsx:252]: remove

**Review by 6YuQing6 — CHANGES_REQUESTED** [review]

File: shopper-app/src/components/SignInButton.tsx (hunk 29 above)
Comment (6YuQing6) [inline shopper-app/src/components/SignInButton.tsx:112]: remove

File: shopper-app/src/views/Account.tsx (hunk 21 above)
Comment (nochoy) [reply shopper-app/src/views/Account.tsx:31]: ^^^

**Review by nochoy — APPROVED** [review]

**Pull request merged**